R/FilteredData-utils.R

```diff
@@ -51,11 +57,32 @@ init_filtered_data.TealData <- function(x, # nolint
   )
 }
 
+#' @keywords internal
+#' @export
+init_filtered_data.teal_data <- function(x, join_keys = teal.data::join_keys(), code = NULL, check = FALSE) { # nolint
+  checkmate::assert_class(code, "CodeClass", null.ok = TRUE)
+  checkmate::assert_class(join_keys, "JoinKeys")
+  checkmate::assert_flag(check)
+  datanames <- x@datanames
+  FilteredData$new(
+    as.list(x@env)[datanames],
+    join_keys = x@join_keys,
+
+    # todo: to be removed soon https://github.com/insightsengineering/teal.slice/issues/471
+    code = teal.data:::CodeClass$new(
+      code = paste(teal.code::get_code(x), collapse = "\n"),
+      dataname = teal.data::get_dataname(x)
+    ),
+    # todo: to remove
+    check = check
+  )
+}
```

Comment:
Looks like `join_keys` is ignored, so the default value is redundant.
I guess `join_keys`, `code` and `check` are only here to maintain consistent arguments between methods?

Reply:
Yes, but this is S3 method so this argument has to be listed. Anyway, don't you think that `init_filtered_data` constructor should be generic only, and `init_filtered_data` should be build from `teal_data` in the internals of teal?

We should give an answer in a near future, when we attempt to include teal_data to the modules. Then we decide what will be the role of FilteredData and whether FD will conduct the code from data loading stage.

Reply:
> Yes, but this is S3 method so this argument has to be listed. 

ok :+1: 

> Anyway, don't you think that `init_filtered_data` constructor should be generic only, and `init_filtered_data` should be build from `teal_data` in the internals of teal?

We won't be needing the multiple methods soon, so it would make sense to have a single function, yes. But then it would be impossible to instantiate `FilteredData` with a list of datasets.

> Then we decide what will be the role of FilteredData

Good point, we are nearing a state where the filter panel has `teal_data` in and `teal_data` out.

Reply:
Maybe passing a note that `join_keys` is ignored and taken from `x@join_keys` would be helpful